### PR TITLE
Add toggle and togglegroup headless and styled components

### PR DESCRIPTION
**Pull request by steffanek**
# What is it?

- [x] Feature / enhancement
- [ ] Bug
- [ ] Docs / tests
- [ ] Other

# Why is it needed?
Those components were missing.

# Checklist:

- [x] My code follows the [developer guidelines of this project](https://github.com/qwikifiers/qwik-ui/blob/main/CONTRIBUTING.md)
- [x] I have performed a self-review of my own code
- [x] I have ran `pnpm change` and documented my changes
- [x] I have add necessary docs (if needed)
- [x] Added new tests to cover the fix / functionality

### CODE REVIEW

**Comment by steffanek** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by thejackshelton** [comment]
I noticed that when there is not a pressed toggle, then the tab order skips the entire component. Is that the intended behavior? Intuitively my thoughts were it would focus the first focusable toggle button.

Here's a preview:

https://github.com/user-attachments/assets/1d990840-1c7c-401b-82dc-56cf98aa092a

**Review by thejackshelton — CHANGES_REQUESTED** [review]

File: apps/website/src/routes/docs/styled/toggle-group/index.mdx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+export const toggleGroupStyledContextId = createContextId<ToggleGroupStyledContext>(
+  'qui-toggle-group-styled',
+);
+
+export type ToggleGroupStyledContext = VariantProps<typeof toggleVariants>;
+
+type ToggleGroupRootProps = PropsOf<typeof HeadlessToggleGroup.Root> &
+  VariantProps<typeof toggleVariants>;
+
+const Root = component$<ToggleGroupRootProps>(({ size, look, ...props }) => {
```
[the remark this answers is not in the document]
Comment (steffanek) [reply apps/website/src/routes/docs/styled/toggle-group/index.mdx:63]: I haven't used inline component.

File: packages/kit-headless/src/components/toggle-group/toggle-group-root.tsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  'bind:value'?: Signal<string[]>;
+};
+
+export type ToggleGroupApiProps = (ToggleGroupSingleProps | ToggleGroupMultipleProps) &
+  ToggleGroupBaseProps &
+  ToggleGroupNavigationProps;
+
+export type ToggleGroupRootProps = PropsOf<'div'> & ToggleGroupApiProps;
+
+export const HToggleGroupRoot = component$<ToggleGroupRootProps>((props) => {
```
[the remark this answers is not in the document]
Comment (steffanek) [reply packages/kit-headless/src/components/toggle-group/toggle-group-root.tsx:89]: I find a workaround to get items in order in CSR land.

File: apps/website/src/routes/docs/styled/toggle/index.mdx [hunk 3]
```diff
@@ -4,8 +4,91 @@ title: Qwik UI | Styled Toggle Component
...
+
+```tsx
+<Toggle>Hello</Toggle
+```
+
+## Usage / Component State
+
+### Initial Value (Uncontrolled)
+
+An initial, uncontrolled value can be provided using the `defaultPressed` prop.
```
Comment (thejackshelton-kunaico) [inline apps/website/src/routes/docs/styled/toggle/index.mdx:79]: We need to update the styled docs here, or remove this and refer to headless

File: packages/kit-headless/src/utils/bound-signal2.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+ * @param initialValue - The initial value to use if no external signal is provided.
+ * @returns A signal that is either bound to the external signal or a new internal signal.
+ *
+ * The returned signal will update the external signal (if provided) whenever its value changes,
+ * and will also update itself when the external signal changes.
+ */
+
+import { createSignal, Signal, useConstant } from '@builder.io/qwik';
+
+export const useBoundSignal = <T,>(
```
Comment (thejackshelton-kunaico) [inline packages/kit-headless/src/utils/bound-signal2.tsx:16]: Let's go ahead and remove this, since this isn't two way data binding

**Review by thejackshelton-kunaico — CHANGES_REQUESTED** [review]

**Pull request merged**